src/Engine/Path.java

```diff
@@ -245,11 +245,11 @@ public void snip(int newLastPoint)
 
   public void snipCollision(GameMap map, Unit unit)
   {
-    for( int i = 0; i < waypoints.size(); i++)
+    for( int i = waypoints.size()-1 ; i > 0; i--)
     {
       PathNode point = waypoints.get(i);
       Unit obstacle = map.getLocation(point.x, point.y).getResident();
-      if( null != obstacle && unit.CO.isEnemy(obstacle.CO) )
+      if( null != obstacle )
```

Comment:
Why remove the enemy-CO condition?

Reply:
Bah, I just realized now that my change probably breaks fog ambushes in some cases.

So. My intent with this change was to snip off anything that would invalidate a move, for its usage in LoadLifecycle below (which should probably also appear in join...)

I guess the big issue here is that *most* of the time, we only want to move into empty spaces... but there are significant and critical exceptions to that logic, and we haven't really codified that anywhere, it's just sorta spread out through the codebase with a bunch of fairly brittle hacks to get stuff to work how it's expected to.

Reply:
> So. My intent with this change was to snip off anything that would invalidate a move

If you move into a transport with an adjacent unit, the move animation goes onto the adjacent unit and you still get the validation failure/WARNING message.

Reply:
Right, which makes it not-a-very-good change.
Which is true largely because of my last paragraph there.

I feel like we need a DRY solution for this.